src/riak_kv_qry_buffers.erl

```diff
@@ -387,27 +396,28 @@ do_delete_qbuf(QBufRef, #state{qbufs = QBufs0,
     case get_qbuf_record(QBufRef, QBufs0) of
         false ->
             {reply, {error, bad_qbuf_ref}, State0};
-        #qbuf{is_ready = false} ->
-            {reply, {error, qbuf_not_ready}, State0};
+        #qbuf{ldb_ref = undefined} ->
+            {reply, ok, State0#state{qbufs = lists:keydelete(QBufRef, 1, QBufs0)}};
```

Comment:
It's for incomplete qbufs (which have not received the complete set of chunks, and are not ready for `fetch_limit`), This actually happens when a query started to be executed but got cancelled. John has hit a bug.